packages/core/components/DirectoryTree/useDirectoryHierarchy.tsx

```diff
@@ -233,12 +214,17 @@ const useDirectoryHierarchy = (
                             sort: sortColumn,
                         });
 
+                        const displayValue =
+                            value === NO_VALUE_NODE
+                                ? `No value ("${hierarchy[depth]}")`
+                                : annotationAtDepth?.getDisplayValue(value) || value;
```

Comment:
Same sort of thing as my earlier comment where I wonder if this would make more sense existing within the `getDisplayValue` function rather than in here.